proxy/logging/LogAccess.cc

```diff
@@ -2631,6 +2631,55 @@ LogAccess::marshal_client_http_transaction_priority_dependence(char *buf)
   return INK_MIN_ALIGN;
 }
 
+/*-------------------------------------------------------------------------
+  -------------------------------------------------------------------------*/
+
+int
+LogAccess::marshal_cache_read_retries(char *buf)
+{
+  if (buf) {
+    int64_t id = 0;
+    if (m_http_sm) {
+      id = m_http_sm->get_cache_sm().get_open_read_tries();
+    }
+    marshal_int(buf, id);
+  }
+  return INK_MIN_ALIGN;
+}
+
+/*-------------------------------------------------------------------------
+  -------------------------------------------------------------------------*/
+
+int
+LogAccess::marshal_cache_write_retries(char *buf)
+{
+  if (buf) {
+    int64_t id = 0;
+    if (m_http_sm) {
+      id = m_http_sm->get_cache_sm().get_open_write_tries();
+    }
+    marshal_int(buf, id);
+  }
+  return INK_MIN_ALIGN;
+}
+
+int
+LogAccess::marshal_cache_collapsed_connection_success(char *buf)
+{
+  if (buf) {
+    int64_t id = 0;
+    if (m_http_sm) {
+      SquidLogCode code = m_http_sm->t_state.squid_codes.log_code;
+      if ((m_http_sm->get_cache_sm().get_open_write_tries() == (m_http_sm->t_state.txn_conf->max_cache_open_write_retries + 1)) &&
```

Comment:
Does it makes sense to set this to may be `-1` when CF was never attempted to distinguish from a failed attempt?

Reply:
👍 yea, gets some more usability out of it

Reply:
I think I may do:
* 0 - not attempted
* -1 - attempted and failed
* 1 - attempted and success

Just like sticking with the negative==failure type returns

Reply:
Seems like it's a bit arguable, but, I agree it's not a big deal as long as it's documented clearly. 👍